habitat-baselines/habitat_baselines/rl/ddppo/policy/resnet_policy.py

```diff
@@ -409,22 +522,39 @@ def __init__(
                 }
             )
 
-        self.visual_encoder = ResNetEncoder(
-            use_obs_space,
-            baseplanes=resnet_baseplanes,
-            ngroups=resnet_baseplanes // 2,
-            make_backbone=getattr(resnet, backbone),
-            normalize_visual_inputs=normalize_visual_inputs,
-        )
+        if backbone == "resnet50" or backbone == "resnet50_imagenet":
+            self.visual_encoder = ResNetEncoder(
+                use_obs_space,
+                baseplanes=resnet_baseplanes,
+                ngroups=resnet_baseplanes // 2,
+                make_backbone=getattr(resnet, backbone),
+                normalize_visual_inputs=normalize_visual_inputs,
+            )
 
-        if not self.visual_encoder.is_blind:
-            self.visual_fc = nn.Sequential(
-                nn.Flatten(),
-                nn.Linear(
-                    np.prod(self.visual_encoder.output_shape), hidden_size
-                ),
-                nn.ReLU(True),
+            if not self.visual_encoder.is_blind:
+                self.visual_fc = nn.Sequential(
+                    nn.Flatten(),
+                    nn.Linear(
+                        np.prod(self.visual_encoder.output_shape), hidden_size
+                    ),
+                    nn.ReLU(True),
+                )
+        elif backbone.startswith("resnet50_clip"):
```

Comment:
Document which options are possible for `backbone`. You should also throw an error if it is an unrecognized option.

Reply:
@ASzot Do you have any thoughts on this?

Reply:
Yes, you should explicitly check for any of them and document in the docstring. For example, if I pass `resnet50_clip_test` it should throw an error that this is not recognized.